### PR TITLE
Baseplate health check v2 take 2

### PR DESCRIPTION
**Pull request by fishy**
Revert is_healthy endpoint defined in BaseplateService, and add health
check v2 into BaseplateServiceV2, and deprecate BaseplateService.

The healthcheck binary is switched to BaseplateServiceV2. It still works
with thrift services using BaseplateService. Added
tests/integration/thrift_healthcheck_tests.py to make sure of that.

### CODE REVIEW

**Comment by fishy** [comment]
👓 @spladug @pacejackson

File: baseplate/thrift/baseplate.thrift [hunk 1]
```diff
@@ -48,7 +69,7 @@ Your service should inherit from this one so that common tools can interact
 with any expected interfaces.
 
 */
-service BaseplateService {
+service BaseplateServiceV2 {
```
Comment (fishy) [inline baseplate/thrift/baseplate.thrift:72]: Or maybe just `BaseplateService2`, not sure which one do y'all prefer.

**Review by pacejackson — APPROVED** [review]
V2 seems fine to me

**Comment by fishy** [comment]
:haircut: Added an integration test to make sure that v2 health check client works with v1 server.

File: tests/integration/thrift_healthcheck_tests.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+
+class GeventPatchedTestCase(unittest.TestCase):
+    def setUp(self):
+        gevent.monkey.patch_socket()
+
+    def tearDown(self):
+        import socket
+
+        reload(socket)
```
Comment (spladug) [inline tests/integration/thrift_healthcheck_tests.py]: this feels like a ton of duplication. could these be parameterized in the original module and either imported here or the test suite below just moved there? (i lean towards the latter as this is just a subset of thrift integration tests)

**Review by spladug — CHANGES_REQUESTED** [review]
looks great. thanks for the tests, that really improves confidence this is non-breaking and continues to be going forward. only thing is making the test suite a little less duplicated there.

**Review by spladug — APPROVED** [review]

**Pull request merged**